src/Attribute/FutureReady.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Attribute;
+
+/**
+ * Marks nodes that are ready to accept a TwigFilter/TwigFunction/TwigTest instead of just a name.
+ */
+#[\Attribute(\Attribute::TARGET_CLASS)]
```

Comment:
should the attribute be applied on the class or on the constructor ? If we make it a method-level attribute, we could remove the special handling of non-overriden constructor (as the inherited constructor would have the attribute)

Reply:
Indeed, looks simpler.